src/service/resource.js

```diff
@@ -320,7 +329,15 @@ export class Resource {
     return this.element.build().then(
       () => {
         this.isBuilding_ = false;
-        this.state_ = ResourceState.NOT_LAID_OUT;
+        // With IntersectionObserver, measure can happen before build
+        // so check if we're "ready for layout" (measured and built) here.
+        if (this.intersect_) {
```

Comment:
Nit:  you don't need to experiment-gate this if you make `hasBeenMeasured()` return false outside the experiment

Reply:
I think it's better to be explicit here rather than hide it a subfunction. In the classical system, resources sequentially follow the phases in `ResourceState` (e.g. build -> measure -> layout) which is somewhat subverted here.

Also `hasBeenMeasured()` is used in many places.

Reply:
whoops, i was mixing up `hasBeenMeasured` and `hasBeenPremeasured`. sg